unblob/handlers/archive/zip.py

```diff
@@ -107,70 +121,79 @@ def is_zip64_eocd(end_of_central_directory: Instance):
-            file, struct.pack("<I", ZIP64_EOCD_LOCATOR_HEADER)
-        ):
-            file.seek(eocd_locator_offset, io.SEEK_SET)
-            eocd_locator = self.cparser_le.zip64_end_of_central_directory_locator_t(
-                file
-            )
-            logger.debug("eocd_locator", eocd_locator=eocd_locator, _verbosity=3)
-
-            # ZIP64 EOCD locator is right before the EOCD record
-            if eocd_locator_offset + len(eocd_locator) == offset:
-                file.seek(start_offset + eocd_locator.offset_of_cd)
-                zip64_eocd = self.cparser_le.zip64_end_of_central_directory_t(file)
-                logger.debug("zip64_eocd", zip64_eocd=zip64_eocd, _verbosity=3)
-
-                if zip64_eocd.signature != ZIP64_EOCD_SIGNATURE:
-                    raise InvalidInputFormat(
-                        "Missing ZIP64 EOCD header record header in ZIP chunk."
-                    )
-                return zip64_eocd
-        raise InvalidInputFormat(
-            "Missing ZIP64 EOCD locator record header in ZIP chunk."
+    def _parse_zip64(self, file: File, eocd_offset: int) -> Tuple[int, Instance]:
+        # ZIP64 EOCD locator is right before the EOCD record
+        eocd_locator_offset = eocd_offset - len(
+            self.cparser_le.zip64_end_of_central_directory_locator_t
         )
+        if eocd_locator_offset < 0:
+            raise InvalidInputFormat("Zip64 offset negative")
+        file.seek(eocd_locator_offset, io.SEEK_SET)
+        eocd_locator = self.cparser_le.zip64_end_of_central_directory_locator_t(file)
+        logger.debug("eocd_locator", eocd_locator=eocd_locator, _verbosity=3)
+        if eocd_locator.signature != ZIP64_EOCD_LOCATOR_SIGNATURE:
+            raise InvalidInputFormat("Zip64 EOCD Locator not found")
+
+        zip64_eocd_offset = eocd_locator_offset - len(
+            self.cparser_le.zip64_end_of_central_directory_t
+        )
+        if zip64_eocd_offset < 0:
+            raise InvalidInputFormat("Zip64 offset negative")
+        file.seek(zip64_eocd_offset, io.SEEK_SET)
+        zip64_eocd = self.cparser_le.zip64_end_of_central_directory_t(file)
+        if zip64_eocd.signature != ZIP64_EOCD_SIGNATURE:
+            raise InvalidInputFormat("Zip64 EOCD not found")
+        return zip64_eocd_offset, zip64_eocd
 
     def calculate_chunk(self, file: File, start_offset: int) -> Optional[ValidChunk]:
+        # start_offset = start of EOCD record
+        # EOCD = end of central directory
+        eocd_offset = start_offset
+        file.seek(eocd_offset, io.SEEK_SET)
+        # from here calculate the beginning of the zip file (needs to interpret the CD offsets)
+        # REASON: EOCD SIGNATURE is much rarer when there are many embedded zip files in a blob
+        # while there are a lot of LOCAL FILE HEADER SIGNATUREs
+
+        end_of_central_directory = self.parse_header(file)
+        assert end_of_central_directory.signature == EOCD_SIGNATURE
```

Comment:
This will always be true since we're matching on `EOCD_SIGNATURE`.

Reply:
Yes, I am aware, that it is always true (now), however it is a bit implicit and that's why it is an assert: previously we matched against something else.

It is not strictly needed, indeed, I have added this assert to keep alive the constant `EOCD_SIGNATURE` (vulture complained for it).

It is harmless, and maybe more of a documentation than a noise, so I would not remove it.